src/runtime/graph_executor/debug/graph_executor_debug.cc

```diff
@@ -300,7 +306,8 @@ class GraphExecutorDebug : public GraphExecutor {
         }
 
         uint32_t eid = entry_id(i, 0);
-        const Device& dev = data_entry_[eid]->device;
+        Device dev = data_entry_[eid]->device;
+        dev.device_type = static_cast<DLDeviceType>(dev.device_type % kRPCSessMask);
```

Comment:
I don't think the device type should be changed within the profiling call. Instead, you should set the device type when the graph executor is created.

Reply:
So the reason that I did this way is that [RPC client](https://github.com/apache/tvm/blob/1c251f50ee616507bdfd8866408e7acf9888cc3f/python/tvm/rpc/client.py#L75) adds an encoding to device type which we need to decode somehow when is used in graph_executor_debugger. I'm not sure if this is the best place to remove encoding.